packages/fxa-payments-server/src/lib/PaymentProvider.ts

```diff
@@ -2,10 +2,16 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-export function isStripe(provider: string | undefined) {
+export type ProviderType = 'paypal' | 'stripe' | 'not_chosen' | undefined;
```

Comment:
If possible, leave the `undefined` out of this type.  I'd prefer to have the object properties to be optional (e.g. `paymentProvider?: ProviderType`) or function parameters to add undefined (e.g. `fn = (p: ProviderType | undefined | null | '') => {}`) if the function want to accept undefined.

Reply:
Okay; I updated this using the former suggestion -- I am not clear on how I'd use the `fn = (p: ProviderType | undefined | null | '') => {}` part -- could you give an example for the future?

Reply:
That was just an example of a function that accepts `ProviderType` as well as some falsey/null-ish values.  The idea is that `ProviderType` is just `'paypal' | 'stripe' | 'not_chosen'` and we can widen the set of accepted value in the parameter declaration instead.